### DISCUSSION TITLE
Special characters in DSN names not supported

### DISCUSSION THREAD

**Discussion by Felix-Braun** [zowe/zowe-explorer-vscode#2132] [Q&A]
**Describe the issue**

We are trying to get a PDS member that has a special character in the DSN name through the ZOWE Explorer in VS Code. In our case the '#' (hash tag character). The connection is made through the APIML using the authentication service. But we are getting the following error:
`Retrieving response from zowe.List Error: z/OSMF REST API Error: Rest API failure with HTTP(S) status 400 <!doctype html><html lang="en"><head><title>HTTP Status 400 – Bad Request</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 400 – Bad Request</h1></body></html>`

We already asked the question in the zowe/api-layer [zowe/api-layer/issues/2728] and were told, that this issue had to be analysed here because if the request is created in a different way when APIML is involved. 

If we can assist you in any way, please let us know.

**To Reproduce**

- open VSCode
- open Zowe explorer
- use a Zowe profile that connects through the authentication service
- search a dataset that contains the special character '#' (hash tag)

**Environment:**

- Windows: Windows 10 (10.0.19044)
- VSCode Verison: 1.72.0
- Zowe Explorer: 2.5.0
- APIML: 2.4
- z/Os: 2.4

**Comment by zFernand0** [comment]
Hey @Felix-Braun,
We believe that this issue might be best if it's resolved by the Zowe NodeJS SDKs.
Please follow this issue/epic for more details in terms of progress:
- https://github.com/zowe/zowe-cli/issues/1633

Once this is done, we will make sure to update this discussion alongside our codebase to integrate with the fixes.

**Reply by JillieBeanSim** [reply]
@zFernand0 I believe we have adopted this, can we close the discussion?

**Reply by zFernand0** [reply]
yes, we have added support for special characters in dataset names.
There is still the PR below:
- #2336 

Maybe we close this discussion once it's merged.
